Orange/widgets/data/owpreprocess.py

```diff
@@ -827,25 +828,34 @@ def __init__(self, parent=None, **kwargs):
                                       "Features",
                                       "Meta data"])
 
-        form.addRow("Randomize:", self.__rand_type_cb)
-        self.layout().addLayout(form)
         self.__rand_type_cb.currentIndexChanged.connect(self.changed)
         self.__rand_type_cb.activated.connect(self.edited)
 
+        self.__rand_seed_ch = QCheckBox()
+        self.__rand_seed_ch.clicked.connect(self.edited)
+
+        form.addRow("Randomize:", self.__rand_type_cb)
+        form.addRow("Replicable shuffling:", self.__rand_seed_ch)
+        self.layout().addLayout(form)
+
     def setParameters(self, params):
         rand_type = params.get("rand_type", Randomize.RandomizeClasses)
         self.__rand_type_cb.setCurrentIndex(rand_type)
+        self.__rand_seed_ch.setChecked(params.get("rand_seed", 1) or 0)
 
     def parameters(self):
-        return {"rand_type": self.__rand_type_cb.currentIndex()}
+        return {"rand_type": self.__rand_type_cb.currentIndex(),
+                "rand_seed": 1 if self.__rand_seed_ch.isChecked() else None}
 
     @staticmethod
     def createinstance(params):
         rand_type = params.get("rand_type", Randomize.RandomizeClasses)
-        return _Randomize(rand_type=rand_type)
+        rand_seed = params.get("rand_seed", 1)
+        return _Randomize(rand_type=rand_type, rand_seed=rand_seed)
 
     def __repr__(self):
-        return self.__rand_type_cb.currentText()
+        return "{}, {}".format(self.__rand_type_cb.currentText(),
```

Comment:
The string constructed in this `__repr__` is not very consistent - it uses human readable names for `rand_type` (not indexes), but the direct representation of `rand_seed` attribute (instead of its meaning).

Not sure if this is used for anything else, but it is definitely not OK for reports, which should be understandable to humans looking at them. Currently this is shown for example:

> Randomize: Classes, 1
> Randomize: Features, None